### ISSUE TITLE
Describe workable network topologies

### ISSUE THREAD

**Issue by ekr** [ietf-wg-add/draft-add-requirements#20]
This document does not seem to describe which topologies this is supposed to work in.

Two specific cases seem like they need attention:

1. Non-upgradeable CPE provided by the carrier.
2. Local non-upgradeable access point/NAT connected to the carrier-provided CPE.

Do we expect that designs should work when I have a client behind each of these?

**Comment by chris-box** [comment]
It's intended that case 1 is covered by the "Unencrypted forwarder" section, but I agree this needs expanding. Essentially given a non-upgradable device there needs to be a way to communicate the associated encrypted resolver that only leverages basic functionality the CPE already has.

Case 2 is not clear to me. A WiFi access point is a layer 2 device so should not be relevant. But do you mean a router that is performing NAT? Are you thinking of something like a smartphone personal hotspot?

**Comment by ekr** [comment]
> Case 2 is not clear to me. A WiFi access point is a layer 2 device so should not be relevant. But do you mean a router that is performing NAT? Are you thinking of something like a smartphone personal hotspot?

Yes, I am assuming a NAT, which in my experience many home wireless APs are.

**Comment by tireddy2** [comment]
Eric is referring to internal CPE (see figure 5 in https://tools.ietf.org/html/draft-btw-add-home-08#section-3.2).

**Comment by chris-box** [comment]
Ok, so case 2 includes any NAT router that is subdividing the local LAN. It's clearly not ideal having two levels of NAT, but given it seems to be prevalent I agree we should support Adaptive DNS Discovery in such an environment.

**Comment by tireddy2** [comment]
https://docs.google.com/document/d/1a9gMyoCIiNrpHXSDIaBASF5vLPH-_KmwvLl5_GLTIo indicates it is not a prevalent deployment !

**Comment by ekr** [comment]
This document does not seem to exist.

**Comment by tireddy2** [comment]
On Thu, 10 Sep 2020 at 18:25, ekr <notifications@github.com> wrote:

> This document does not seem to exist.
>

Try
https://docs.google.com/document/d/1a9gMyoCIiNrpHXSDIaBASF5vLPH-_KmwvLl5_GLTIoo/



>
>
> On Thu, Sep 10, 2020 at 12:29 AM Tirumaleswar Reddy <

**Comment by ekr** [comment]
This is a long document. Which section/page do you believe makes this claim?
